pkg/bootstrap/bootstrap_plain_git.go

```diff
@@ -390,3 +422,31 @@ func (b *PlainGitBootstrapper) ReportComponentsHealth(ctx context.Context, insta
 	b.logger.Successf("all components are healthy")
 	return nil
 }
+
+func getOpenPgpEntity(keyRing openpgp.EntityList, passphrase, keyID string) (*openpgp.Entity, error) {
+	if len(keyRing) == 0 {
+		return nil, fmt.Errorf("empty GPG key ring")
+	}
+
+	var entity *openpgp.Entity
+	if keyID != "" {
+		for _, ent := range keyRing {
+			if ent.PrimaryKey.KeyIdString() == keyID {
```

Comment:
This should be case insensitive, we also should trim the `0x` prefix from keyID.